pipeline_reviews/transform.py

```diff
@@ -0,0 +1,92 @@
+"""Validates received review inputs"""
+
+from datetime import datetime
+from os import environ
+
+import pandas as pd
+from pandas import DataFrame
+from dotenv import load_dotenv
+from psycopg2.extras import RealDictCursor
+from psycopg2 import connect
+from psycopg2.extensions import connection
+
+
+def get_db_connection() -> connection:
+    """Returns PSQL database connection"""
+    load_dotenv()
+    return connect(dbname=environ["DATABASE_NAME"],
+                    user=environ["DATABASE_USERNAME"],
+                    password=environ["DATABASE_PASSWORD"],
+                    cursor_factory=RealDictCursor)
+
+
+def get_release_date(game_id: int, conn: connection, cache: dict) -> datetime:
+    """Retrieves the release date for a game with a provided ID"""
+    if game_id in cache.keys():
+        return cache[game_id]
+    else:
+        # TODO get data from connection
+        cache[game_id] = 3# TODO value from above
+        return cache[game_id]
+
+
+def correct_playtime(reviews_df: DataFrame) -> DataFrame:
+    """Returns a data-frame with valid playtime recordings only"""
```

Comment:
I think the way this function is written would make it hard to test... would it be possible to work out the release data as another function and keep this as only the validating the correct playtime?

Reply:
release date is calculated in a separate function :)